python/llm/example/transformers/transformers_int4/falcon/README.md

```diff
@@ -0,0 +1,99 @@
+# Falcon
+
+In this directory, you will find examples on how you could apply BigDL-LLM INT4 optimizations on Falcon models. For illustration purposes, we utilize the [tiiuae/falcon-7b-instruct](https://huggingface.co/tiiuae/falcon-7b-instruct) and [tiiuae/falcon-40b-instruct](https://huggingface.co/tiiuae/falcon-40b-instruct) as reference Falcon models.
+
+## 0. Requirements
+To run these examples with BigDL-LLM, we have some recommended requirements for your machine, please refer to [here](../README.md#recommended-requirements) for more information.
+
+## Example: Predict Tokens using `generate()` API
+In the example [generate.py](./generate.py), we show a basic use case for a Falcon model to predict the next N tokens using `generate()` API, with BigDL-LLM INT4 optimizations.
+### 1. Install
+We suggest using conda to manage environment:
+```bash
+conda create -n llm python=3.9
+conda activate llm
+
+pip install bigdl-llm[all] # install bigdl-llm with 'all' option
+pip install einops # additional package required for falcon-7b-instruct and falcon-40b-instruct to conduct generation
+```
+
+### 2. (Optional) Download Model and Replace File
+If you select the Falcon models [tiiuae/falcon-7b-instruct](https://huggingface.co/tiiuae/falcon-7b-instruct) or [tiiuae/falcon-40b-instruct](https://huggingface.co/tiiuae/falcon-40b-instruct), please note that there is a bug in their model implementation (`modelling_RW.py`) that causes failures when using the KV cache. Consequently, the BigDL-LLM INT4 optimization may not achieve optimal performance with these models.
+
+To address this issue, we have provided two files: [src/falcon-7b-instruct/modelling_RW.py](./src/falcon-7b-instruct/modelling_RW.py) and [src/falcon-40b-instruct/modelling_RW.py](./src/falcon-40b-instruct/modelling_RW.py). You can replace the `modelling_RW.py file` in the `tiiuae/falcon-7b-instruct` or `tiiuae/falcon-40b-instruct` model files respectively with these files. By doing so, you can resolve the KV cache bug and obtain optimal performance from the BigDL-LLM INT4 optimization.
+
+#### 2.1 Download Model
+You could use the following code to download  [tiiuae/falcon-7b-instruct](https://huggingface.co/tiiuae/falcon-7b-instruct) or [tiiuae/falcon-40b-instruct](https://huggingface.co/tiiuae/falcon-40b-instruct) with a specific snapshot id. Please note that the provided patched `modelling_RW.py` files are specifically based on these commits.
```

Comment:
Please note that the `modelling_RW.py` files that we provide are  based on these specific commits